### ISSUE TITLE
[google_sign_in] flutter build appbundle --release throws PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)

### ISSUE THREAD

**Issue by verscph** [flutter/flutter#75224]
Hi,

I configured google_sign_in 4.5.3 successfully for a Flutter app running on a firebase project (dev) and deployed it to Google Play Internal Testing using:
- the package name be.mycompany.myapp
- with SHA fingerprints registered in the firebase app
- google-services.json added to my flutter android project
- OAuth APIs enabled

Everything works as expected.

Today, I'm preparing an Alpha Closed testing version of the app that connects to another firebase app (production) using:
- the same package name be.mycompany.myapp as I can only upload apps with that package name to the Play Store
- with new SHA fingerprints registered in the firebase app as SHA keys can't be shared between 2 different firebase projects using the same package name
- the new google-services.json was added to my flutter android project
- OAuth APIs enabled for the new firebase project (prod)

When running the app in release mode using Android Studio, everything works as expected.

When I upload the release version to Google Play, I can login using username/password, but using google_sign-in raises the error: 
`PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)`
which means there's an issue with the SHA fingerprints. As all is working fine in release mode running via Android Studio, I don't know what happened between running the build locally and uploading it to Google Play.

I tried adding SHA keys, both on the firebase project and OAuth API, but the restriction that the same SHA key and package combination can only be used once causes the release version in Google Play can't make use of google_sign_in.

Any ideas on how to fix this?

Thanks!

**Comment by ameysunu** [comment]
Hey,

Have you tried checking this link at [Stackoverflow](https://stackoverflow.com/questions/54557479/flutter-and-google-sign-in-plugin-platformexceptionsign-in-failed-com-google)?

**Comment by darshankawar** [comment]
@verscph 
There's similar issue open describing your case, https://github.com/flutter/flutter/issues/33393 and check the underlying comment / solution and see if it works for you https://github.com/flutter/flutter/issues/33393#issuecomment-510395178.

**Comment by verscph** [comment]
Yes, I checked the Stackoverflow post you mention and also this [Medium post](https://medium.com/@mttibor/flutter-firebase-auth-google-sign-in-android-release-mode-67256b432f25).

Following this post's advice, I managed to get the local (Android Studio) environment working. I added the SHA-1 key to the dev environment and the SHA-256 to production to get Firebase auth working on the Google Play internal testing and Closed Testing. Like that, I can login to the correct Firebase projects, but google_sign_in still doesn't work on Closed Testing. Of course, using the SHA-1 App Siging key for one environment and the SHA-256 App Signing key for another environment is not the way to go, but ok... I got it working for now.

I figured out that running the release version via Android Studio generates an APK. Uploading that APK
`flutter build apk --release` to Google Play Closed Closed Testing makes google_sign_in work while uploading the
`flutter build appbundle --release` build to Closed Testing gives
`PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)`

Maybe this last info helps you detecting what goes wrong with signing and keys?

**Comment by verscph** [comment]
@darshankawar tx, but I don't see a OAuth consent screen menu:
![Screenshot 2021-02-03 at 08 10 03](https://user-images.githubusercontent.com/34537296/106712522-5c9d3200-65f9-11eb-8801-495a44ca9c26.jpg)
and I'm sure I gave the consent, filled in support email, ...

I'm pretty sure this has something to do with the different release build types (apk, appbundle) that get signed using other keys.

**Comment by ameysunu** [comment]
Can you try checking out this article as well https://stackoverflow.com/a/53663789/11602890? I think this should be the one causing you the error, since already ``flutter build apk --release`` works for you.

**Comment by darshankawar** [comment]
@verscph 
There was similar issue reported related to `google_sign_in` not working using appbundle, https://github.com/flutter/flutter/issues/41909 and this SO link seems to have resolved the issue, https://github.com/flutter/flutter/issues/41909#issuecomment-548931910. (same as shared above).

**Comment by verscph** [comment]
@darshankawar Tx, moving the App Signing key certificate from the dev firebase to prod firebase did the trick, but still... this can't be the right way to get things working? As you can't use the same SHA fingerprint on 2 different firebase projects with the same package name, this means that you can only log in using your Google Account on 1 environment, dev OR release OR prod?

**Comment by darshankawar** [comment]
@verscph 
I agree. The link seems to be more of a workaround rather than a solution and should also be mentioned in the plugin documentation.

**Comment by MahendraRaj94** [comment]
In my case this solution worked. 
I had enabled Automatic App signing enabled in play console which generates its own SHA key.
**Adding that SHA key to firebase project where I had production application worked for me**

**Comment by vkmaxcooldude** [comment]
I just added the SHA-256 Key along with SHA-1 Key to my Firebase Android Project from Google Play Developers Console and it worked out pretty well.

**Comment by solserer-labs** [comment]
I was not using firebase so it was hard to find a solution but I found one in a youtube comment..

You have to create a Web Application OAuth 2.0 Client Id in the google console and pass that clientId to the GoogleSignIn constructor instead of an android one.

**Comment by aymericsgnth** [comment]
> I was not using firebase so it was hard to find a solution but I found one in a youtube comment..
> 
> You have to create a Web Application OAuth 2.0 Client Id in the google console and pass that clientId to the GoogleSignIn constructor instead of an android one.

You're right.... And it's not written in the documentation.

**Comment by andvalsol** [comment]
@aymericsgnth can you please tell me how to accomplish that?

**Comment by lewinpauli** [comment]
I am having the same problem, App works in Apple App Store, in Android Debug mode, flutter build apk --release, flutter build appbundle --release

But not in playstore

I added the playstore sha keys to firebase but after multiple weeks nothing seems to work...

@solserer-labs  can you maybe provide a screenshot of what you mean exactly? That would be really nice

**Comment by piyushrathodIn** [comment]
> I was not using firebase so it was hard to find a solution but I found one in a youtube comment..
> You have to create a Web Application OAuth 2.0 Client Id in the google console and pass that clientId to the GoogleSignIn constructor instead of an android one.

@solserer-labs, will it work on iOS too?

**Comment by bimba-digital** [comment]
Aquí la solución:
https://www.youtube.com/watch?v=jxA--phMORk

**Comment by jerbio** [comment]
Hello,

Has any one found the solution to this problem. The issue with this bug is specific to releasing to the play store and not using firebase. Most Api Exceptions 10 codes are related to Sha-1 fingerprint mismatches. However as the original posted noted the apk build works without issues. 
Troubleshooting steps: 
- I extracted and installed the apk from the aab using buildtools and the sign in works on the device.
- I compared the Sha-1 fingerprint from the Apk build and the AAB build, they both match, so we know this isn't a Sha1 local build issue.

Is there an issue with using this package on the google play store without firebase?

**Comment by jerbio** [comment]
Hello,

Finally figured this one out, the issue is still related to sha-1 Certifcate but however the problem is you need to create a new andorid oauth client-id for android using the Sha-1 finger print of the google play console app signing certificate .

Go to your app page on the 
Google Play Console -> [Your Application] -> Setup -> App Integrity.

Click on "App Signing Tab"

Under this page you should see there are two sections "App signing key certificate" and "Upload key certificate"
![Screenshot 2023-07-05 at 2 34 51 AM](https://github.com/flutter/flutter/assets/4109175/5daa4bc1-7535-4b2b-ac12-b3c0072de1ec)
You need to copy the sha-1 fingerprint Under "App signing key certificate"

Go to your google cloud console.
Create a new Oauth client id for android:
![Screenshot 2023-07-05 at 2 41 28 AM](https://github.com/flutter/flutter/assets/4109175/5250e14a-2b5a-4948-8242-686a9a94c41b)

Paste the Sha-1 key in the newly created Android Oauth client-id

![Screenshot 2023-07-05 at 2 44 11 AM](https://github.com/flutter/flutter/assets/4109175/c78e1a67-b7a6-4d8c-bfe7-3d459e7582d0)


Note none of this required Firebase integration.  Hope this cracks it for every one.

**Comment by lewinpauli** [comment]
@jerbio thank you for your detailed explanaition, sadly it does still not work for me

I am using firebase, and the app works perfectly fine on ios and appstore, also in debug and release mode on android, but when I push the app to playstore I get the following error:

PlatformException(sign_in_failed, com.google.android.gms.common.api.b:10: , null, null)

I already deleted the whole android profile in firebase, reconfigured the app manually and with flutterfire cli
nothing seems to work

I also added both App Signing Keys (SHA1 & SHA256) to firebase, still no change


Does anyone else have a solution?

**Comment by therjtkumar** [comment]
i have fixed this issue by following ways:

- First if you using goole_sign_in with firebase make sure to add (SHA1 & SHA256) to firebase using your **upload signing keystore** (if you uploaded you app to playstore you can also find this in playstore as **upload key certificate** section under app signing), or you can generate the (SHA1 & SHA256) using terminal by targeting your upload signing keystore. Follow this for the reference: https://developers.google.com/android/guides/client-auth

- One important thing to remember that SHA1 should also be uploaded to google cloud project in your project (in **credentials** page under **OAuth 2.0 Client IDs** section). You can verify the same otherwise add it manually (in my case firebase did it for me as my google cloud project was linked with firbase).

- Finally the ultimate goal here is to match the (SHA1 & SHA256) in firebase with the app which signed using your upload keystore (not debug.keystore). Thats how i have fixed this issue for app release.

Hope this helps !

**Comment by EhtashamNasir** [comment]
I had same issue, but that Fixed when it had latest keys and then it had legacy keys, those Legacy Keys added to firebase, And Without rebuild or Google Services.json update it worked
![image](https://github.com/flutter/flutter/assets/96730912/56645dff-dbe5-4664-9647-9d36f6641854)
![image](https://github.com/flutter/flutter/assets/96730912/588b659a-649a-4b05-8dbb-589ec9cb71fb)

**Comment by muccy-timeware** [comment]
I write here a follow-up explaining how we fixed the problem after months of trials and failures:

- we created a web (not Android) OAuth key on Google Cloud Console;
- we used Flutterfire CLI to register and download configuration files (so we switched Firebase initialization to use the generated .dart options file).
    - This caused `apply plugin: 'com.google.firebase.crashlytics'` to be added inside `app/build.gradle` file;
    - and `classpath 'com.google.firebase:firebase-crashlytics-gradle:2.8.1'` inside `build.gradle` file.
- we upgraded GMS:
    - `distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip` in `gradle-wrapper.properties`
    - `classpath 'com.android.tools.build:gradle:7.3.0'`, `classpath 'com.google.gms:google-services:4.4.0'`  in `build.gradle` file.

Magically, uploading the release AAB to Play Store internal testing track, Google Sign In started to work.

PS: I find literally infuriating and ridiculous that a Google product works better on iOS than Android. This is not even a solution. This is a pure combination of factors that solved the problem by magic

**Comment by ragummrsa2930** [comment]
Team, I'm also facing this issue. In My case, Below is the issue I'm getting.

[google_sign_in] throws PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)

Based on chatgpt answer 

The error message you provided indicates that the Google Sign-In process failed due to an ApiException with code 10. ApiException 10 typically corresponds to the error code DEVELOPER_ERROR.

Here are a few common reasons why this error might occur and some potential solutions:
Invalid SHA-1 or SHA-256 fingerprint in Firebase console, Missing or incorrect OAuth 2.0 client ID, Incorrect package name, and Other configuration issues. 

Here My problem is Congifuration issues. 

I have 2 Firebase base project 
1. Test (com.companyname.app.dev)
2. Prod(com.company name.app)

While integrating Firebase into the Flutter app, we will use Flutterfire configure comment. Sometimes this comment registers package names with different project IDs, so that time like the Oauth2 client ID auto-generated and added to the Google project. Go to [google project](https://console.cloud.google.com/apis/credentials?project=) and search package names. like the production project has only the production app package name and the Test project has only the test app package name.  If anything is wrongly auto-generated please delete the key and rerun the project. 

In my case, this is the problem and it is working fine for me.

**Comment by guplem** [comment]
This worked like a charm for me: https://medium.com/codebrew/flutter-google-sign-in-without-firebase-3680713966fb
In summary: I created new OAuth 2.0 Client IDs using the SHA1 keys from the command `./gradlew signInReport`.

**Comment by GlaucoMendes** [comment]
> I had same issue, but that Fixed when in Google Play App Signing had latest keys and then it had legacy keys, those Legacy Keys added to firebase, And Without rebuild or Google Services.json update it worked ![image](https://private-user-images.githubusercontent.com/96730912/271893418-56645dff-dbe5-4664-9647-9d36f6641854.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNTQ1NTMsIm5iZiI6MTcxNDM1NDI1MywicGF0aCI6Ii85NjczMDkxMi8yNzE4OTM0MTgtNTY2NDVkZmYtZGJlNS00NjY0LTk2NDctOWQzNmY2NjQxODU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAxMzA1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYjk2NjAzN2QyMWQyNGZiOTA4MDcyMGY5MjM1NGVjYTA5YTJlMjY1Yjk2MjY3NjRiZGZhZmNjODM2NDk5N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eJlSe9ZRQm0nTVRiiuLXpika6v3q2pMJI0BJlbavY54) ![image](https://private-user-images.githubusercontent.com/96730912/271893594-5f86a7fd-70ea-476f-a6a1-ae56bd32123d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNTQ1NTMsIm5iZiI6MTcxNDM1NDI1MywicGF0aCI6Ii85NjczMDkxMi8yNzE4OTM1OTQtNWY4NmE3ZmQtNzBlYS00NzZmLWE2YTEtYWU1NmJkMzIxMjNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAxMzA1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyZjdmNjZlNTA0YWE1NzMxZDQ5OWNkNjRiNjQ1NzFkZTdkYTlhYTJmYjEwYmJiNmI1MGQ3YjMyZmIyZDhkY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.529D-oYaGCCJ5L-LPN1eyPWZfCMToK3VywEipiZVXRE)

I have to come here and thank you. After 5 years as a Flutter developer, I never experienced this problem, it was the first time. It took 4 days trying to resolve it and this was the solution. Thank you very much

**Comment by M4rcoosV** [comment]
> Eu tive o mesmo problema, mas isso foi corrigido quando no Google Play App Signing tinha as chaves mais recentes e depois tinha chaves legadas, essas chaves legadas foram adicionadas ao Firebase e, sem reconstruir ou atualizar o Google Services.json, funcionou !

eu consegui resolver meu problema após mais de 7 dias gerando chaves SHA1 mais finalmente esse comentário livrou meu problema, tentem isso por favor se estiver com o mesmo problema ao gerar o aplicativo na play store ou AAB .

**Comment by LamNguyen17** [comment]
Step 1: ./gradlew signingReport or gradlew signingReport
Step 2: Paste SHA1 & SHA256 generate by command line to Firebase
Step 3: Download & replace the google-service.json
Step 4: flutter clean & rebuild app

**Comment by Nde-Dilan** [comment]
I recently ran into the same issue while developing my app. I started off in debug mode and, following the docs, I added the SHA-1 and SHA-256 keys to the Firebase project settings. Everything was working fine so far. But when I switched to the release version to see how my app behaves in that environment, I started encountering errors.

I tried a bunch of different solutions, including directly adding the key to Google Cloud 
![GCP interface to add restrictions to API keys](https://github.com/user-attachments/assets/51e485db-c260-46fa-bb30-efb9807d7885)
, but none of that worked. After some more troubleshooting, I decided to try adding the key back to Firebase's project settings. And to my surprise, that fixed the issue! Now, everything works as expected in both debug and release mode.
 
![Firebase project settings](https://github.com/user-attachments/assets/aca8d237-923f-4969-b167-8dccdfc21cba)
 As you can see I have 4 keys now(2 for debugging and 2 for release)

So, if you're facing similar issues, make sure you've added both SHA-1 and SHA-256 keys to Firebase's project settings. That might just do the trick!

**Comment by Alt-WN** [comment]
We have similar issues:
[https://github.com/flutter/flutter/issues/160302](url) - closed
[https://github.com/flutter/flutter/issues/110789](url) - closed
[https://github.com/flutter/flutter/issues/137614](url)
[https://github.com/flutter/flutter/issues/33393](url)
After spending a week trying to recreate new sample projects (limited only to "logging on with Google") from scratch on 3 different developmnet computers: Windows 10, Windows 11 and macOS Sequoia 15.2 with I believe all documented steps done carefully and all above suggested "solutions" with failure (I still hope that maybe I missed one of them...), I think that root cause is that message in "ApiException: 10" is **too general**. There are too many reasons that may cause this problem. The message in "ApiException: 10" should be more precise to narrow areas of investigation.

**Comment by Tox46** [comment]
Following up on the comment from @jshji252 , if you have a Legacy App Signing Key, add the fingerprints of that on Firebase too, even if you asked a key upgrade.

**Comment by stuartmorgan** [comment]
Closing as a duplicate of https://github.com/flutter/flutter/issues/33393.
